dynatrace/environment_v2/service_level_objectives.py

```diff
@@ -0,0 +1,282 @@
+        :param time_from: The start of the requested timeframe.
+        :param time_to: The end of the requested timeframe.
+
+        :returns Slo: the requested SLO
+        """
+        params = {
+            "from": time_from,
+            "to": time_to,
+        }
+        response = self.__http_client.make_request(path=f"{self.ENDPOINT}/{slo_id}", params=params).json()
+        return Slo(raw_element=response)
+
+    def create(
+        self,
+        name: str,
+        target: float,
+        warning: float,
+        timeframe: str,
+        use_rate_metric: Optional[bool] = False,
+        metric_rate: Optional[str] = None,
+        metric_numerator: Optional[str] = None,
+        metric_denominator: Optional[str] = None,
+        evaluation_type: Optional[str] = "AGGREGATE",
+        filter_: Optional[str] = None,
+        custom_description: Optional[str] = None,
+        enabled: Optional[bool] = False,
+    ) -> "Response":
+        """Creates a new SLO.
+
+        :param name: The name of the SLO.
+        :param target: The target value of the SLO.
+        :param warning: The warning value of the SLO. At warning state the SLO is still fulfilled but is getting close to failure.
+        :param timeframe: The timeframe for the SLO evaluation. Use the syntax of the global timeframe selector.
+        :param use_rate_metric: The type of the metric to use for SLO calculation - an existing percentage-based metric (true) or a ratio of two metrics (false)
+        :param metric_rate: The percentage-based metric for the calculation of the SLO. Required when the useRateMetric is set to true.
+        :param metric_numerator: The metric for the count of successes (the numerator in rate calculation).Required when the useRateMetric is set to false.
+        :param metric_denominator: The total count metric (the denominator in rate calculation). Required when the useRateMetric is set to false.
+        :param evaluation_type: The evaluation type of the SLO.
+        :param filter_: The entity filter for the SLO evaluation. Use the syntax of entity selector.
+        :param custom_description: The custom description of the SLO.
+        :param enabled: The SLO is enabled (true) or disabled (false).
+
+        :returns Response: HTTP response for the request
+        """
+        params = {
+            "name": name,
+            "target": target,
+            "warning": warning,
+            "timeframe": timeframe,
+            "useRateMetric": use_rate_metric,
+            "metricRate": metric_rate,
+            "metricNumerator": metric_numerator,
+            "metricDenominator": metric_denominator,
+            "evaluationType": evaluation_type,
+            "filter": filter_,
+            "customDescription": custom_description,
+            "enabled": enabled,
+        }
+        return self.__http_client.make_request(path=self.ENDPOINT, method="POST", params=params)
+
+    def update(
```

Comment:
I don't like this implementation. the update (should be called `put`) method should simply receive an `Slo` instance, and you can have a separated `create` method to help the user create one if they chose to do so manually, that method can have the same parameters as this one currently has (they can also chose to do a `get`  themselves, and later call `put` with the result of the `get` 

The Slo instance also needs to have a Slo.put() method to simplify this flow.

Example, this should be possible

```python
    slo = dt.slos.get("my-slo")  # or slo = dt.slos.create(params)
    slo.name = "new name"
    slo.put() # or # dt.slos.put(slo)
    
    local_slo = dt.slos.create(params)  # This does not call post and definitely does not do a hidden get, it returns an empty local SLO object
    local_slo.post() # or  #  dt.slos.post(local_slo)
```

Reply:
Re-implemented this whole part. Functions are now called `get`, `post`, `put`, `delete`. `post` and `put` only take Slo object. `create` function added to help user create Slo object from scratch. Added `put` and `post` inside the Slo class too.
I have left the `post` & `put` also in the SloService, but they now seem redundant.
This is all now supported:
```python
all_slos = dt.slos.list()                               # GET list of all SLOs in tenant
one_slo = dt.slos.get(slo_id=list(all_slos)[0].id)      # GET SLO from Dynatrace with all details

one_slo.name = "New Name"
dt.slos.put(one_slo)                                    # PUT via SloService
dt.slos.post(
    dt.slos.create(                                     # POST via SloService
        target=80.5, 
        warning=90, 
        use_rate_metric=False, 
        metric_numerator="ext:metric.numerator", 
        metric_denominator="ext:metric.denominator"
    )
)
my_new_slo = dt.slos.create(                            # local SLO (object) creation
    target=90.5, 
    warning=95, 
    timeframe="-2h", 
    use_rate_metric=True, 
    metric_rate="builtin:some.metric.rate"
)
my_new_slo.post()                                       # POST via local object
my_new_slo.description = "Forgot to add description"
my_new_slo.put()                                        # PUT via lcoal object
```
Do you think there is a reason for keeping post and put within SloService?

Reply:
Thanks, this looks beautiful, I don't know if there is a good reason but I think it is good for consistency, ie when people type `dt.slos.` it automatically completes with all possible operations, instead of them having to read the docs/examples and understand that the Slo object has the post and put methods